### PR TITLE
Add Minimesos v0.10.2

### PR DESCRIPTION
**Pull request by vlal-zz**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/minimesos.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  homepage "https://minimesos.org/"
+  url "https://raw.githubusercontent.com/ContainerSolutions/minimesos/0.10.2/bin/minimesos"
+  sha256 "58d05e2574e1c82b8f2f366f92d6dc464438603bd0fb5fb4b6383b6123233af0"
+
+  def install
+    bin.install "minimesos"
+  end
+
+  test do
+    system "minimesos", "version"
```
Comment (MikeMcQuaid) [inline Formula/minimesos.rb]: Test is failing, seems to need `depends_on "docker"`: https://bot.brew.sh/job/Homebrew%20Core/9364/version=any_sierra/testReport/junit/brew-test-bot/sierra/test_minimesos/

Also, use `system "#{bin}/minimesos"`

File: Formula/minimesos.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Minimesos < Formula
+  desc "Testing infrastructure for Mesos frameworks"
+  homepage "https://minimesos.org/"
+  url "https://raw.githubusercontent.com/ContainerSolutions/minimesos/0.10.2/bin/minimesos"
```
Comment (MikeMcQuaid) [inline Formula/minimesos.rb]: Would be better to use `https://github.com/ContainerSolutions/minimesos/archive/0.10.2.tar.gz` and just `bin.install "bin/minimesos"` if needed.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/minimesos.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class Minimesos < Formula
+  desc "Testing infrastructure for Mesos frameworks"
+  homepage "https://minimesos.org/"
+  url "https://github.com/ContainerSolutions/minimesos/archive/0.10.2.tar.gz"
+  sha256 "16b198078c9384426f8849b08e92e173816c4d6626331bc6569aa5de89dffbdd"
+
+  unless system "docker ps -aq > /dev/null"
+    depends_on "docker"
```
Comment (MikeMcQuaid) [inline Formula/minimesos.rb]: Please just use `depends_on "docker"`. If you wish to use your own Docker please consider adding a `:docker`/`DockerRequirement` to Homebrew/brew.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/minimesos.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "16b198078c9384426f8849b08e92e173816c4d6626331bc6569aa5de89dffbdd"
+
+  depends_on "docker"
+
+  def install
+    bin.install "bin/minimesos"
+  end
+
+  test do
+    system "test -e #{bin}/minimesos && true || false"
```
Comment (MikeMcQuaid) [inline Formula/minimesos.rb]: Please run the executable and `assert_match` on the version or something, thanks.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/minimesos.rb (hunk 4 above)
Comment (vlal) [inline Formula/minimesos.rb]: Apologies, I ran into issues trying that, minimesos requires not only the docker client but the docker server running as well. The `depends_on "docker"` only installs the client which is useless without the server. I tried going the docker-machine virtualbox route but that required virtualbox which is in homebrew-cask and which doesn't play nice within the confines of the test environment. Lastly I tried docker-machine-xhyve but that required root permission which wasn't a viable option either. Not sure what my next steps should be, should this be in homebrew-cask or is there another option?

Comment (vlal) [inline Formula/minimesos.rb]: @MikeMcQuaid ping?

Comment (MikeMcQuaid) [inline Formula/minimesos.rb]: I'm on vacation so my responses may take a while. `system "#{bin}/minimesos", "version"` doesn't work?

Comment (MikeMcQuaid) [inline Formula/minimesos.rb]: `assert_match "minimesos", shell_output("#{bin}/minimesos --help")` should work. If not, assert_match "minimesos", shell_output("#{bin}/minimesos --help 2>&1") should.

File: Formula/minimesos.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "16b198078c9384426f8849b08e92e173816c4d6626331bc6569aa5de89dffbdd"
+
+  depends_on "docker"
+
+  def install
+    bin.install "bin/minimesos"
+  end
+
+  test do
+    assert_match "Minimesos requires Docker 1.11.0 or higher", shell_output("#{bin}/minimesos")
```
Comment (MikeMcQuaid) [inline Formula/minimesos.rb]: Test is failing: https://bot.brew.sh/job/Homebrew%20Core/10466/version=any_sierra/testReport/junit/brew-test-bot/sierra/test_minimesos/

Perhaps just `assert_match "Docker", shell_output("#{bin}/minimesos 2>&1")` is the best bet?

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/minimesos.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "16b198078c9384426f8849b08e92e173816c4d6626331bc6569aa5de89dffbdd"
+
+  depends_on "docker"
+
+  def install
+    bin.install "bin/minimesos"
+  end
+
+  test do
+    assert_match "Docker", shell_output("#{bin}/minimesos --help 2>&1")
```
Comment (MikeMcQuaid) [inline Formula/minimesos.rb]: `assert_match "Docker", shell_output("#{bin}/minimesos --help 2>&1", 125)` and we should be home free.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `minimesos 0.10.2 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Pull request merged**